Remove import/export profiles and revert file menu items in file menu

---
 .../files/browser/fileActions.contribution.ts |  45 +----
 .../browser/userDataProfile.ts                | 162 +-----------------
 2 files changed, 4 insertions(+), 203 deletions(-)

Index: third-party-src/src/vs/workbench/contrib/files/browser/fileActions.contribution.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/files/browser/fileActions.contribution.ts
+++ third-party-src/src/vs/workbench/contrib/files/browser/fileActions.contribution.ts
@@ -10,7 +10,7 @@ import { MenuId, MenuRegistry, registerA
 import { ICommandAction } from '../../../../platform/action/common/action.js';
 import { KeyMod, KeyCode } from '../../../../base/common/keyCodes.js';
 import { openWindowCommand, newWindowCommand } from './fileCommands.js';
-import { COPY_PATH_COMMAND_ID, REVEAL_IN_EXPLORER_COMMAND_ID, OPEN_TO_SIDE_COMMAND_ID, REVERT_FILE_COMMAND_ID, SAVE_FILE_COMMAND_ID, SAVE_FILE_LABEL, SAVE_FILE_AS_COMMAND_ID, SAVE_FILE_AS_LABEL, SAVE_ALL_IN_GROUP_COMMAND_ID, OpenEditorsGroupContext, COMPARE_WITH_SAVED_COMMAND_ID, COMPARE_RESOURCE_COMMAND_ID, SELECT_FOR_COMPARE_COMMAND_ID, ResourceSelectedForCompareContext, OpenEditorsDirtyEditorContext, COMPARE_SELECTED_COMMAND_ID, REMOVE_ROOT_FOLDER_COMMAND_ID, REMOVE_ROOT_FOLDER_LABEL, SAVE_FILES_COMMAND_ID, COPY_RELATIVE_PATH_COMMAND_ID, SAVE_FILE_WITHOUT_FORMATTING_COMMAND_ID, SAVE_FILE_WITHOUT_FORMATTING_LABEL, OpenEditorsReadonlyEditorContext, OPEN_WITH_EXPLORER_COMMAND_ID, NEW_UNTITLED_FILE_COMMAND_ID, NEW_UNTITLED_FILE_LABEL d, OpenEditorsSelectedFileOrUntitledContext } from './fileConstants.js';
+import { COPY_PATH_COMMAND_ID, REVEAL_IN_EXPLORER_COMMAND_ID, OPEN_TO_SIDE_COMMAND_ID, SAVE_FILE_COMMAND_ID, SAVE_FILE_LABEL, SAVE_FILE_AS_COMMAND_ID, SAVE_FILE_AS_LABEL, SAVE_ALL_IN_GROUP_COMMAND_ID, OpenEditorsGroupContext, COMPARE_WITH_SAVED_COMMAND_ID, COMPARE_RESOURCE_COMMAND_ID, SELECT_FOR_COMPARE_COMMAND_ID, ResourceSelectedForCompareContext, OpenEditorsDirtyEditorContext, COMPARE_SELECTED_COMMAND_ID, REMOVE_ROOT_FOLDER_COMMAND_ID, REMOVE_ROOT_FOLDER_LABEL, SAVE_FILES_COMMAND_ID, COPY_RELATIVE_PATH_COMMAND_ID, SAVE_FILE_WITHOUT_FORMATTING_COMMAND_ID, SAVE_FILE_WITHOUT_FORMATTING_LABEL, OpenEditorsReadonlyEditorContext, OPEN_WITH_EXPLORER_COMMAND_ID, NEW_UNTITLED_FILE_COMMAND_ID, NEW_UNTITLED_FILE_LABEL, OpenEditorsSelectedFileOrUntitledContext } from './fileConstants.js';
 import { CommandsRegistry, ICommandHandler } from '../../../../platform/commands/common/commands.js';
 import { ContextKeyExpr, ContextKeyExpression } from '../../../../platform/contextkey/common/contextkey.js';
 import { KeybindingsRegistry, KeybindingWeight } from '../../../../platform/keybinding/common/keybindingsRegistry.js';
@@ -20,7 +20,7 @@ import { CLOSE_SAVED_EDITORS_COMMAND_ID,
 import { AutoSaveAfterShortDelayContext } from '../../../services/filesConfiguration/common/filesConfigurationService.js';
 import { WorkbenchListDoubleSelection } from '../../../../platform/list/browser/listService.js';
 import { Schemas } from '../../../../base/common/network.js';
-import { DirtyWorkingCopiesContext, EnterMultiRootWorkspaceSupportContext, HasWebFileSystemAccess, WorkbenchStateContext, WorkspaceFolderCountContext, SidebarFocusContext, ActiveEditorCanRevertContext, ActiveEditorContext, ResourceContextKey, ActiveEditorAvailableEditorIdsContext, MultipleEditorsSelectedInGroupContext, TwoEditorsSelectedInGroupContext, SelectedEditorsInGroupFileOrUntitledResourceContextKey } from '../../../common/contextkeys.js';
+import { DirtyWorkingCopiesContext, EnterMultiRootWorkspaceSupportContext, HasWebFileSystemAccess, WorkbenchStateContext, WorkspaceFolderCountContext, SidebarFocusContext, ActiveEditorContext, ResourceContextKey, ActiveEditorAvailableEditorIdsContext, MultipleEditorsSelectedInGroupContext, TwoEditorsSelectedInGroupContext, SelectedEditorsInGroupFileOrUntitledResourceContextKey } from '../../../common/contextkeys.js';
 import { IsWebContext } from '../../../../platform/contextkey/common/contextkeys.js';
 import { ServicesAccessor } from '../../../../platform/instantiation/common/instantiation.js';
 import { ThemeIcon } from '../../../../base/common/themables.js';
@@ -244,12 +244,6 @@ appendToCommandPalette({
 });
 
 appendToCommandPalette({
-	id: REVERT_FILE_COMMAND_ID,
-	title: nls.localize2('revert', "Revert File"),
-	category: Categories.File
-});
-
-appendToCommandPalette({
 	id: COMPARE_WITH_SAVED_COMMAND_ID,
 	title: nls.localize2('compareActiveWithSaved', "Compare Active File with Saved"),
 	category: Categories.File,
@@ -352,26 +346,6 @@ MenuRegistry.appendMenuItem(MenuId.OpenE
 
 MenuRegistry.appendMenuItem(MenuId.OpenEditorsContext, {
 	group: '2_save',
-	order: 20,
-	command: {
-		id: REVERT_FILE_COMMAND_ID,
-		title: nls.localize('revert', "Revert File"),
-		precondition: OpenEditorsDirtyEditorContext
-	},
-	when: ContextKeyExpr.and(
-		// Not: editor groups
-		OpenEditorsGroupContext.toNegated(),
-		// Not: readonly editors
-		OpenEditorsReadonlyEditorContext.toNegated(),
-		// Not: untitled editors (revert closes them)
-		ResourceContextKey.Scheme.notEqualsTo(Schemas.untitled),
-		// Not: auto save after short delay
-		AutoSaveAfterShortDelayContext.toNegated()
-	)
-});
-
-MenuRegistry.appendMenuItem(MenuId.OpenEditorsContext, {
-	group: '2_save',
 	order: 30,
 	command: {
 		id: SAVE_ALL_IN_GROUP_COMMAND_ID,
@@ -682,21 +656,6 @@ MenuRegistry.appendMenuItem(MenuId.Menub
 	},
 	order: 1
 });
-
-MenuRegistry.appendMenuItem(MenuId.MenubarFileMenu, {
-	group: '6_close',
-	command: {
-		id: REVERT_FILE_COMMAND_ID,
-		title: nls.localize({ key: 'miRevert', comment: ['&& denotes a mnemonic'] }, "Re&&vert File"),
-		precondition: ContextKeyExpr.or(
-			// Active editor can revert
-			ContextKeyExpr.and(ActiveEditorCanRevertContext),
-			// Explorer focused but not on untitled
-			ContextKeyExpr.and(ResourceContextKey.Scheme.notEqualsTo(Schemas.untitled), FoldersViewVisibleContext, SidebarFocusContext)
-		),
-	},
-	order: 1
-});
 
 MenuRegistry.appendMenuItem(MenuId.MenubarFileMenu, {
 	group: '6_close',
Index: third-party-src/src/vs/workbench/contrib/userDataProfile/browser/userDataProfile.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/userDataProfile/browser/userDataProfile.ts
+++ third-party-src/src/vs/workbench/contrib/userDataProfile/browser/userDataProfile.ts
@@ -152,8 +152,6 @@ export class UserDataProfilesWorkbenchCo
 		this.registerProfilesActions();
 		this._register(this.userDataProfilesService.onDidChangeProfiles(() => this.registerProfilesActions()));
 
-		this._register(this.registerExportCurrentProfileAction());
-
 		this.registerCreateFromCurrentProfileAction();
 		this.registerNewProfileAction();
 		this.registerDeleteProfileAction();
@@ -386,34 +384,6 @@ export class UserDataProfilesWorkbenchCo
 		return disposables;
 	}
 
-	private registerExportCurrentProfileAction(): IDisposable {
-		const that = this;
-		const disposables = new DisposableStore();
-		const id = 'workbench.profiles.actions.exportProfile';
-		disposables.add(registerAction2(class ExportProfileAction extends Action2 {
-			constructor() {
-				super({
-					id,
-					title: localize2('export profile', "Export Profile..."),
-					category: PROFILES_CATEGORY,
-					f1: true,
-				});
-			}
-
-			async run() {
-				const editor = await that.openProfilesEditor();
-				editor?.selectProfile(that.userDataProfileService.currentProfile);
-			}
-		}));
-		disposables.add(MenuRegistry.appendMenuItem(MenuId.MenubarShare, {
-			command: {
-				id,
-				title: localize2('export profile in share', "Export Profile ({0})...", that.userDataProfileService.currentProfile.name),
-			},
-		}));
-		return disposables;
-	}
-
 
 	private registerCreateFromCurrentProfileAction(): void {
 		const that = this;
Index: third-party-src/src/vs/workbench/contrib/userDataProfile/browser/userDataProfilesEditor.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/userDataProfile/browser/userDataProfilesEditor.ts
+++ third-party-src/src/vs/workbench/contrib/userDataProfile/browser/userDataProfilesEditor.ts
@@ -223,13 +223,7 @@ export class UserDataProfilesEditor exte
 					const actions: IAction[] = [];
 					if (this.templates.length) {
 						actions.push(new SubmenuAction('from.template', localize('from template', "From Template"), this.getCreateFromTemplateActions()));
-						actions.push(new Separator());
 					}
-					actions.push(toAction({
-						id: 'importProfile',
-						label: localize('importProfile', "Import Profile..."),
-						run: () => this.importProfile()
-					}));
 					return actions;
 				}
 			},
@@ -294,50 +288,9 @@ export class UserDataProfilesEditor exte
 		if (templateActions.length) {
 			actions.push(new SubmenuAction('from.template', localize('new from template', "New Profile From Template"), templateActions));
 		}
-		actions.push(new Separator());
-		actions.push(toAction({
-			id: 'importProfile',
-			label: localize('importProfile', "Import Profile..."),
-			run: () => this.importProfile()
-		}));
 		return actions;
 	}
 
-	private async importProfile(): Promise<void> {
-		const disposables = new DisposableStore();
-		const quickPick = disposables.add(this.quickInputService.createQuickPick());
-
-		const updateQuickPickItems = (value?: string) => {
-			const quickPickItems: IQuickPickItem[] = [];
-			if (value) {
-				quickPickItems.push({ label: quickPick.value, description: localize('import from url', "Import from URL") });
-			}
-			quickPickItems.push({ label: localize('import from file', "Select File...") });
-			quickPick.items = quickPickItems;
-		};
-
-		quickPick.title = localize('import profile quick pick title', "Import from Profile Template...");
-		quickPick.placeholder = localize('import profile placeholder', "Provide Profile Template URL");
-		quickPick.ignoreFocusOut = true;
-		disposables.add(quickPick.onDidChangeValue(updateQuickPickItems));
-		updateQuickPickItems();
-		quickPick.matchOnLabel = false;
-		quickPick.matchOnDescription = false;
-		disposables.add(quickPick.onDidAccept(async () => {
-			quickPick.hide();
-			const selectedItem = quickPick.selectedItems[0];
-			if (!selectedItem) {
-				return;
-			}
-			const url = selectedItem.label === quickPick.value ? URI.parse(quickPick.value) : await this.getProfileUriFromFileSystem();
-			if (url) {
-				this.createNewProfile(url);
-			}
-		}));
-		disposables.add(quickPick.onDidHide(() => disposables.dispose()));
-		quickPick.show();
-	}
-
 	async createNewProfile(copyFrom?: URI | IUserDataProfile): Promise<void> {
 		await this.model?.createNewProfile(copyFrom);
 	}
Index: third-party-src/src/vs/workbench/contrib/userDataProfile/browser/userDataProfilesEditorModel.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/userDataProfile/browser/userDataProfilesEditorModel.ts
+++ third-party-src/src/vs/workbench/contrib/userDataProfile/browser/userDataProfilesEditorModel.ts
@@ -977,28 +977,7 @@ export class UserDataProfilesEditorModel
 			() => this.userDataProfileManagementService.switchProfile(profileElement.profile)
 		));
 
-		const copyFromProfileAction = disposables.add(new Action(
-			'userDataProfile.copyFromProfile',
-			localize('copyFromProfile', "Duplicate..."),
-			ThemeIcon.asClassName(Codicon.copy),
-			true, () => this.createNewProfile(profileElement.profile)
-		));
-
-		const exportAction = disposables.add(new Action(
-			'userDataProfile.export',
-			localize('export', "Export..."),
-			ThemeIcon.asClassName(Codicon.export),
-			true,
-			() => this.userDataProfileImportExportService.exportProfile(profile)
-		));
 
-		const deleteAction = disposables.add(new Action(
-			'userDataProfile.delete',
-			localize('delete', "Delete"),
-			ThemeIcon.asClassName(Codicon.trash),
-			true,
-			() => this.removeProfile(profileElement.profile)
-		));
 
 		const newWindowAction = disposables.add(new Action(
 			'userDataProfile.newWindow',
@@ -1012,12 +991,6 @@ export class UserDataProfilesEditorModel
 		primaryActions.push(activateAction);
 		primaryActions.push(newWindowAction);
 		const secondaryActions: IAction[] = [];
-		secondaryActions.push(copyFromProfileAction);
-		secondaryActions.push(exportAction);
-		if (!profile.isDefault) {
-			secondaryActions.push(new Separator());
-			secondaryActions.push(deleteAction);
-		}
 
 		const profileElement = disposables.add(this.instantiationService.createInstance(UserDataProfileElement,
 			profile,
@@ -1094,17 +1067,10 @@ export class UserDataProfilesEditorModel
 				() => this.previewNewProfile(cancellationTokenSource.token)
 			));
 			secondaryActions.push(previewProfileAction);
-			const exportAction = disposables.add(new Action(
-				'userDataProfile.export',
-				localize('export', "Export..."),
-				ThemeIcon.asClassName(Codicon.export),
-				isUserDataProfile(copyFrom),
-				() => this.exportNewProfile(cancellationTokenSource.token)
-			));
 			this.newProfileElement = disposables.add(this.instantiationService.createInstance(NewProfileElement,
 				copyFrom,
 				[primaryActions, secondaryActions],
-				[[cancelAction], [exportAction]],
+				[[cancelAction], []],
 			));
 			const updateCreateActionLabel = () => {
 				if (createAction.enabled) {
@@ -1123,7 +1089,6 @@ export class UserDataProfilesEditorModel
 				}
 				if (e.name || e.copyFrom) {
 					updateCreateActionLabel();
-					exportAction.enabled = isUserDataProfile(this.newProfileElement?.copyFrom);
 				}
 			}));
 			disposables.add(this.userDataProfilesService.onDidChangeProfiles((e) => {
